packages/fluentui/react-northstar/src/components/Attachment/AttachmentAction.tsx

```diff
@@ -22,16 +38,30 @@ const AttachmentAction = compose<
 >(Button, {
   className: attachmentActionClassName,
   displayName: 'AttachmentAction',
-}) as ComponentWithAs<'button', AttachmentActionProps> & { create: ShorthandFactory<any> };
+  overrideStyles: true,
+}) as ComponentWithAs<'button', AttachmentActionProps> & { shorthandConfig: ShorthandConfig<AttachmentActionProps> };
 
 AttachmentAction.defaultProps = {
   accessibility: buttonBehavior,
   as: 'button',
-  iconOnly: true,
-  text: true,
 };
-AttachmentAction.propTypes = Button.propTypes;
+AttachmentAction.propTypes = {
+  ...commonPropTypes.createCommon({
+    content: 'shorthand',
+  }),
+  disabled: PropTypes.bool,
+  icon: customPropTypes.shorthandAllowingChildren,
+  iconPosition: PropTypes.oneOf(['before', 'after']),
+  loader: customPropTypes.itemShorthandWithoutJSX,
+  loading: PropTypes.bool,
+  onClick: PropTypes.func,
+  onFocus: PropTypes.func,
+  primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+  secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
```

Comment:
Do we really support `loader`, `loading`, `primary`, `secondary`, etc.?

Reply:
I left the styles necessary for them.. I thought we should leave these, as loading and laoder are generic enough, and we have primary and secondary everywhere, so it would be awkward not to have them here..